src/dhcp/aca_dhcp_server.cpp

```diff
@@ -197,4 +211,175 @@ int ACA_Dhcp_Server::_get_db_size() const
   }
 }
 
+void ACA_Dhcp_Server::_init_dhcp_msg_ops()
+{
+  _parse_dhcp_msg_ops[DHCP_MSG_NONE] = _parse_dhcp_none;
+  _parse_dhcp_msg_ops[DHCP_MSG_DHCPDISCOVER] = _parse_dhcp_discover;
+}
+
+void ACA_Dhcp_Server::dhcps_recv(void *message)
+{
+  const dhcp_message *dhcpmsg = nullptr;
+  uint8_t msg_type = 0;
+
+  if (nullptr == message) {
+    ACA_LOG_ERROR("DHCP message is null!\n");
+    return;
+  }
+
+  dhcpmsg = (dhcp_message *)message;
+
+  if (_validate_dhcp_message(dhcpmsg)) {
+    ACA_LOG_ERROR("Invalid DHCP message!\n");
+    return;
+  }
+
+  msg_type = _get_message_type(dhcpmsg);
+  _parse_dhcp_msg_ops[msg_type](dhcpmsg);
+
+  return;
+}
+
+void ACA_Dhcp_Server::dhcps_xmit(void *message)
+{
```

Comment:
@cj-chung this is where our DHCP server would send back the DHCP offer packet, we will need to provide the interface for this code to call.

Reply:
The interface for packet_out in ACA is: 
`void ACA_OVS_Control::packet_out(const char *bridge, const char *opt) `

I can make bridge default to br-tun or br-int.
For 'opt', it accepts a char string like this:
` "in_port=controller packet=02AC10FF002202AC10FF001108004500001C000100000A015A9DAC10FF0BAC10FF160800F7FF00000000 actions=normal"`

'actions' is optional, the other two are required. 
 'packet' is a hex string of the crafted packet.